crates/primitives/src/transaction/mod.rs

```diff
@@ -456,8 +552,19 @@ impl Compact for Transaction {
                 (Transaction::Eip1559(tx), buf)
             }
             3 => {
-                let (tx, buf) = TxEip4844::from_compact(buf, buf.len());
-                (Transaction::Eip4844(tx), buf)
+                let identifier = buf.get_u8() as usize;
+                match identifier {
```

Comment:
why are we grabbing the identifier again here? would appreciate a comment

Reply:
We tried to leave an explanation above the function declaration [here](https://github.com/anton-rs/op-reth/blob/clabby/op-reth/crates/primitives/src/transaction/mod.rs#L537-L539)
```rust
    // For backwards compatibility purposes, only 2 bits of the type are encoded in the identifier
    // parameter. In the case of a 3, the full transaction type is read from the buffer as a
    // single byte.
    fn from_compact(mut buf: &[u8], identifier: usize) -> (Self, &[u8]) {
```

But will also add a comment in this branch. (if you think the function comment is actually sufficient, you can close my follow-up pr :) )

Reply:
Addressed in https://github.com/anton-rs/op-reth/pull/104. I think it's actually nice to have this clarifying comment as well so leaning towards including it for posterity 😄